chatgpt_wrapper/browser.py

```diff
@@ -0,0 +1,108 @@
+import shutil
+import uuid
+import json
+import asyncio
+import re
+from typing import Optional
+from playwright.async_api import async_playwright
+from playwright._impl._api_structures import ProxySettings
+from playwright.async_api._generated import BrowserType,BrowserContext
+from .log import LogCapable
+
+class AsyncBrowser(LogCapable):
+    # A single play can launch many browsers
+    play=None
+    def __init__(self):
+        self.browser:BrowserType=None
+        self.browser_context:BrowserContext=None
+        self.user_data_dir=None
+        self.page=None
+        self.session={}
+
+    async def create(self,browser:str='firefox', headless:bool=True, proxy: Optional[ProxySettings] = None, debug_log=None):
+        super().__init__(debug_log=debug_log)
+        if AsyncBrowser.play is None:
+            AsyncBrowser.play=await async_playwright().start()
+
+        if not hasattr(self.play,browser):
+            self.log.error("Browser %s is invalid, falling back on firefox. ",browser)
+            browser='firefox'
+        self.browser:BrowserType=getattr(self.play,browser)
+
+        try:
+            self.browser_context=await self.browser.lauch_persistent_context(
```

Comment:
Spelling error.

Reply:
Sorry about my poor English but I can't see any spelling error here...
And this part code runs on my setup (after fixing bugs somewhere else).